wpilibNewCommands/src/main/native/include/frc2/command/CommandRobot.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <frc/TimedRobot.h>
+#include <frc/smartdashboard/SendableChooser.h>
+#include <units/time.h>
+
+#include "frc2/command/CommandPtr.h"
+#include "frc2/command/CommandScheduler.h"
+#include "frc2/command/Commands.h"
+
+namespace frc2 {
+/**
+  CommandRobot is a wrapper over TimedRobot designed to work well with Command
+ * based programming.
+ *
+ * <p>The CommandRobot class is intended to be subclassed by a user when
+ creating a
+ * robot program.
+ */
+class CommandRobot : public frc::TimedRobot {
+ public:
+  void RobotInit() final;
+
+  void RobotPeriodic() final;
+
+  void AutonomousInit() final;
+
+  void AutonomousPeriodic() final;
+
+  void AutonomousExit() final;
+
+  void TeleopInit() final;
+
+  void TeleopPeriodic() final;
+
+  void TeleopExit() final;
+
+  void DisabledInit() final;
+
+  void DisabledPeriodic() final;
+
+  void DisabledExit() final;
```

Comment:
Which of the iterative methods do we want to mark as final? [This comment](https://github.com/wpilibsuite/allwpilib/issues/6624#issuecomment-2119008904) says that `robotPeriodic()` and `teleopPeriodic()` could be marked final to force teams to use `addPeriodic()` and commands, and I could see the case for marking `robotInit()` as final as another part of prepping for deprecating `robotInit()`, but I don't see any reasoning for marking `autonomousPeriodic()`, `teleopInit()`, `teleopPeriodic()`, `teleopExit()`, `disabledInit()`, `disabledPeriodic()`, and `disabledExit()` as final. Maybe with [the addition of `RobotModeTriggers`](https://github.com/wpilibsuite/allwpilib/pull/5902) last year, we want to encourage people to use those `Trigger`s instead of the robot mode functions?

Reply:
I would argue proper use of the Commands framework would have no usage of periodic functions, except when the user explicitly opts into doing it for a reason they have decided (eg addPeriodic for updating sensor values).

I can see more of an argument for leaving some of the *Init functions particularly autonomousInit but I think a similar situation still applies with the *Init functions in that there are very specific things they need to do and they shouldn't do more than that when using Commands.